rollup-cli/tools/slasher.test.js

```diff
@@ -0,0 +1,127 @@
+        feePlan: bb.feePlan.length ? bb.feePlan : [0, 0],
+        compressedTx: `0x${bb.getDataAvailable().toString("hex")}`,
+        offChainHash: bb.getOffChainHash().toString(),
+        nTxPerToken: bb.getCountersOut().toString(),
+        beneficiary: beneficiary
+    };
+}
+
+function manageEvent(event) {
+    if (event.event == "OnChainTx") {
+        const txData = rollupUtils.decodeTxData(event.args.txData);
+        return {
+            fromIdx: txData.fromId,
+            toIdx: txData.toId,
+            amount: txData.amount,
+            loadAmount: BigInt(event.args.loadAmount),
+            coin: txData.tokenId,
+            ax: BigInt(event.args.Ax).toString(16),
+            ay: BigInt(event.args.Ay).toString(16),
+            ethAddress: BigInt(event.args.ethAddress).toString(),
+            onChain: true
+        };
+    }
+}
+
+contract("RollupPoS", async (accounts) => {
+
+    let insPoseidonUnit;
+    let insRollupTest;
+    let insVerifier;
+    let insRollupPoS;
+
+    let urlNode = "http://localhost:8545";
+    let addressSC;
+    const  walletPath = "./walletSlash.json";
+    const  pass = "foo";
+    let abi;
+    let walletJson;
+    let rollupDB;
+    let db;
+    const nLevels = 24;
+
+    async function forgeBlock(events = undefined) {
+       
+        const block = await rollupDB.buildBatch(maxTx, nLevels);
+        if (events) {
+            events.forEach(elem => {
+                block.addTx(manageEvent(elem));
+            });
+        }
+        await block.build();
+      
+        const inputSm = buildInputSm(block, beneficiary);
+        await insRollupTest.forgeBatchTest(inputSm.oldStateRoot, inputSm.newStateRoot, inputSm.newExitRoot,
+            inputSm.onChainHash, inputSm.feePlan, inputSm.compressedTx, inputSm.offChainHash, inputSm.nTxPerToken,
+            inputSm.beneficiary);
+          
+        await rollupDB.consolidate(block);
+     
+    }
+  
+    const maxTx = 10;
+    const maxOnChainTx = 10;
+    const {
+        0: owner,
+        1: beneficiary,
+    } = accounts;
+
+    before(async () => {
+        // Deploy poseidon
+        const C = new web3.eth.Contract(poseidonUnit.abi);
+        insPoseidonUnit = await C.deploy({ data: poseidonUnit.createCode() })
+            .send({ gas: 2500000, from: owner });
+
+        // Deploy Verifier
+        insVerifier = await Verifier.new();
+
+        // Deploy Rollup test
+        insRollupTest = await RollupTest.new(insVerifier.address, insPoseidonUnit._address,
+            maxTx, maxOnChainTx);
+
+        insRollupPoS = await RollupPoS.new(insRollupTest.address);
+
+        addressSC = insRollupPoS.contract._address;
+        abi = insRollupPoS.abi;
+        walletJson = JSON.parse(fs.readFileSync(walletPath, "utf8"));
+
+        db = new SMTMemDB();
+        rollupDB = await RollupDB(db);
+        
+    });
+    it("slash", async () => {
+        for(let i = 0; i <1011; i++) {
+            await forgeBlock();
```

Comment:
we could use 
https://github.com/iden3/rollup/blob/master/test/contracts/helpers/timeTravel.js
to add blocks xD